src/store/modules/auth/actions/index.js

```diff
@@ -33,24 +34,35 @@ export const authFailure = error => ({
   }
 });
 
-export const authAction = (userData, history, url = '') => async dispatch => {
+export const authAction = ({
+  userData,
+  history,
+  url = undefined
+}) => async dispatch => {
+  const { username = undefined, email, password } = userData;
   dispatch(authPending());
   try {
+    const authRoute = username ? 'signup' : 'login';
+
+    const details = { email, password } &&
+      username && { email, password, username };
+
     const response = await axios({
       method: 'post',
-      baseURL: `${process.env.apiUrl}/auth/login`,
-      data: userData
+      url: `${config.apiUrl}/auth/${authRoute}`,
+      data: details
     });
     const { token } = response.data.data;
+
     saveToLocalStorage({ token, url });
-    const user = decodeToken({ history });
 
-    Toastr.success('Welcome to ErrorSwag');
+    const user = decodeToken({ history });
+    Toastr.success('Welcome to ErrorSwag!');
     dispatch(authSuccess(user));
 
-    return url ? history.push(url) : history.push();
-  } catch (error) {
-    const { message } = error.response.data;
+    return url ? history.push(url) : history.push('/articles');
+  } catch ({ response }) {
+    const { message } = response.data;
```

Comment:
instead of removing these lines 
```js
    } catch (error) {
    const { message } = error.response.data;
```, I suggest you use an `if` statement there to also cater for the error coming from `error.response.data` , incase a user encounters a `network error`.

Reply:
Oh True. However, In the case of a network failure, `axios` still returns a response since that it is what was stated on the try block. 
Hence, I think 
``` const message = response && response.data && response.data.message ```
might be okay. What do you think?